src/hooks/swap/useWrapCallback.tsx

```diff
@@ -43,19 +43,26 @@ export default function useWrapCallback(): UseWrapCallbackReturns {
     [inputCurrency, amount]
   )
 
-  const callback = useMemo(() => {
-    return async () => {
-      if (!parsedAmountIn) return
-      switch (wrapType) {
-        case TransactionType.WRAP:
-          return wrappedNativeCurrencyContract?.deposit({ value: `0x${parsedAmountIn.quotient.toString(16)}` })
-        case TransactionType.UNWRAP:
-          return wrappedNativeCurrencyContract?.withdraw(`0x${parsedAmountIn.quotient.toString(16)}`)
-        case undefined:
-          return undefined
-      }
+  const wrapCallback = useCallback(async (): Promise<WrapTransactionInfo | UnwrapTransactionInfo | void> => {
+    if (!parsedAmountIn || !wrappedNativeCurrencyContract) return
+    switch (wrapType) {
+      case TransactionType.WRAP:
+        return {
+          response: await wrappedNativeCurrencyContract.deposit({ value: `0x${parsedAmountIn.quotient.toString(16)}` }),
+          type: TransactionType.WRAP,
+          amount: parsedAmountIn,
+        }
+      case TransactionType.UNWRAP:
+        return {
+          response: await wrappedNativeCurrencyContract.withdraw(`0x${parsedAmountIn.quotient.toString(16)}`),
+          type: TransactionType.WRAP,
+          amount: parsedAmountIn,
+        }
+      case undefined:
+        return undefined
     }
   }, [parsedAmountIn, wrappedNativeCurrencyContract, wrapType])
+  const callback = usePerfEventHandler('onWrapSend', wrapCallback, parsedAmountIn && { amount: parsedAmountIn })
```

Comment:
`{ amount: parsedAmountIn }` is inputs to `wrapCallback` right? kinda hard to tell without comments in `usePerfEventHandler` and without labelling these inputs

Reply:
refactored to (hopefully) make this more clear. `usePerfEventHandler` is meant to mirror the arguments of the perf event handler, so I want it to be ordinal inputs (ie not an options object).

Reply:
this makes sense now thanks